src/lib/firebase.js

```diff
@@ -18,17 +19,43 @@ var firebaseConfig = {
 let fb = firebase.initializeApp(firebaseConfig);
 let db = firebase.firestore();
 
-export function updatePurchaseDate(collectionName, itemId = {}) {
+export function updatePurchaseDate(collectionName, itemId) {
   const itemRef = db.collection(collectionName).doc(itemId);
-  itemRef.update({
-    purchaseDates: firebase.firestore.FieldValue.arrayUnion(
-      firebase.firestore.Timestamp.now().toMillis(),
-    ),
-  });
+  return itemRef
+    .update({
+      purchaseDates: firebase.firestore.FieldValue.arrayUnion(
+        firebase.firestore.Timestamp.now().toMillis(),
+      ),
+    })
+    .then(function() {
+      updateFrequency(collectionName, itemId);
+    });
+}
+
+export function updateFrequency(collectionName, itemId) {
```

Comment:
I don't fully understand the firebase methods used in this function, but I don't think there needs to be two separate update functions (updatePurchaseDates, updateFrequency). Since we're getting the purchaseDates as soon as we open the app in `App.js`, I think we could do the checks you're doing here in `List.js`, and then we could pass collectionName, itemId and updatedFrequency all in one update function.  That way, we wouldn't have to communicate multiple times with Firebase for one transaction.  Let me know if this is unclear!